src/main/scala/org/broadinstitute/dsde/rawls/model/AttributeUpdateOperations.scala

```diff
@@ -0,0 +1,63 @@
+package org.broadinstitute.dsde.rawls.model
+
+import com.wordnik.swagger.annotations.{ApiModelProperty, ApiModel}
+import org.broadinstitute.dsde.rawls.model.WorkspaceJsonSupport._
+import spray.json._
+
+import scala.annotation.meta.field
+
+/**
+ * Created by mbemis on 7/23/15.
+ */
+object AttributeUpdateOperations {
+  sealed trait AttributeUpdateOperation
+  case class AddUpdateAttribute(attributeName: String, addUpdateAttribute: Attribute) extends AttributeUpdateOperation
+  case class RemoveAttribute(attributeName: String) extends AttributeUpdateOperation
+  case class AddListMember(attributeListName: String, newMember: Attribute) extends AttributeUpdateOperation
+  case class RemoveListMember(attributeListName: String, removeMember: Attribute) extends AttributeUpdateOperation
+
+  private val AddUpdateAttributeFormat = jsonFormat2(AddUpdateAttribute)
+  private val RemoveAttributeFormat = jsonFormat1(RemoveAttribute)
+  private val AddListMemberFormat = jsonFormat2(AddListMember)
+  private val RemoveListMemberFormat = jsonFormat2(RemoveListMember)
+
+  implicit object AttributeUpdateOperationFormat extends RootJsonFormat[AttributeUpdateOperation] {
+
+    override def write(obj: AttributeUpdateOperation): JsValue = {
+      val json = obj match {
+        case x: AddUpdateAttribute => AddUpdateAttributeFormat.write(x)
+        case x: RemoveAttribute => RemoveAttributeFormat.write(x)
+        case x: AddListMember => AddListMemberFormat.write(x)
+        case x: RemoveListMember => RemoveListMemberFormat.write(x)
+      }
+
+      JsObject(json.asJsObject.fields + ("op" -> JsString(obj.getClass.getSimpleName)))
+    }
+
+    override def read(json: JsValue) : AttributeUpdateOperation = json match {
+      case JsObject(fields) =>
+        val op = fields.getOrElse("op", throw new DeserializationException("missing op property"))
+        op match {
+          case JsString("AddUpdateAttribute") => AddUpdateAttributeFormat.read(json)
+          case JsString("RemoveAttribute") => RemoveAttributeFormat.read(json)
+          case JsString("AddListMember") => AddListMemberFormat.read(json)
+          case JsString("RemoveListMember") => RemoveListMemberFormat.read(json)
+          case x => throw new DeserializationException("unrecognized op: " + x)
+        }
+
+      case _ => throw new DeserializationException("unexpected json type")
+    }
+  }
+
+  @ApiModel(value = "Entity update definition")
+  case class EntityUpdateDefinition(
+                        @(ApiModelProperty@field)(required = true, value = "The name of the entity")
+                        name: String,
+                        @(ApiModelProperty@field)(required = true, value = "The type of the entity")
+                        entityType: String,
+                        @(ApiModelProperty@field)(required = true, value = "The update operations of the entity")
```

Comment:
That's accurate. I'm actually completely re-working how Swagger is used so I won't fix that in this PR but I'll make sure to update that in the Swagger update PR.